app/graphql/mutations/profile/associate_systems.rb

```diff
@@ -24,15 +24,51 @@ def resolve(args = {})
 
       def find_profile(profile_id)
         ::Pundit.authorize(
-          context[:current_user],
+          current_user,
           ::Profile.find(profile_id),
           :edit?
         )
       end
 
       def find_hosts(system_ids)
-        ::Pundit.policy_scope(context[:current_user], ::Host)
-                .where(id: system_ids)
+        existing_systems = ::Pundit.policy_scope(current_user, ::Host)
+                                   .where(id: system_ids)
+        save_hosts(system_ids - existing_systems.pluck(:id))
```

Comment:
I guess you do this just to avoid a few unnecessary requests to the inventory?

Reply:
Right - no need to find from inventory and save a host we already have in our DB.